core/internal/filetransfer/file_transfer_manager.go

```diff
@@ -16,9 +16,16 @@ const (
 	DefaultConcurrencyLimit = 128
 )
 
+type FileTransferType string
+
+const (
+	FileTransferTypeDefault = FileTransferType("Default")
+)
```

Comment:
instead of a map, how about a struct with a field for each file transfer type?

```go
// FileTransfers is a collection of file transfers by upload destination type.
type FileTransfers struct {
	// Default makes an HTTP PUT request to the destination URL with the file contents.
	Default FileTransfer
)
```

then you can even have a method to select the correct one:

```go
// For returns the appropriate FileTransfer for the Task.
func (ft *FileTransfers) For(task *Task) FileTransfer {
	return ft.Default
}
```

don't forget documentation comments!

Reply:
added suggestions!